### PR TITLE
Comparing two denotations doesn't work properly

### PR DESCRIPTION
**Pull request by laughedelic**
I saw again assertion-related code in scarph and rememberd that we still have this problem. [Here](https://github.com/ohnosequences/cosas/blob/master/src/test/scala/cosas/RecordTests.scala#L71-L72) for example.

### CODE REVIEW

**Comment by laughedelic** [comment]
@eparejatobes review, please.

File: src/main/scala/cosas/types/syntax.scala [hunk 1]
```diff
@@ -13,6 +13,13 @@ case object syntax {
     final def ==>[S <: AnyType](s: S): T ==> S = new (T ==> S)(tpe,s)
   }
 
+  final case class DenotationSyntax[T <: AnyType, V <: T#Raw](val v: V) extends AnyVal {
+
+    def =~=[W <: T#Raw](w: T := W): Boolean = v == w.value
```
Comment (eparejatobes) [inline src/main/scala/cosas/types/syntax.scala]: Why the type param here?

**Comment by laughedelic** [comment]
ok. merging?

**Comment by eparejatobes** [comment]
yes!

**Pull request merged**